File: src/main/java/com/nicomincuzzi/frameworkless/domain/MazeMap.java
package com.nicomincuzzi.frameworkless.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class MazeMap {

    @JsonProperty
    private List<Room> rooms;

    public List<Room> getRooms() {
        return rooms;
    }
}

File: src/main/kotlin/com/nicomincuzzi/frameworkless/domain/MazeMap.kt
package com.nicomincuzzi.frameworkless.domain

import com.fasterxml.jackson.annotation.JsonProperty

class MazeMap {
    @JsonProperty val rooms: List<Room> = listOf()

    fun getRoomById(roomId: Int): Room? {
        for (room in rooms) {
            if (room.id == roomId) return room
        }
        return null
    }

    fun getArrayRooms(): List<Room> {
        return rooms
    }
}